src/requests/cache/sorted_set/sorted_set_put_elements.rs

```diff
@@ -1,17 +1,104 @@
+use std::collections::HashMap;
+
 use momento_protos::cache_client::{SortedSetElement, SortedSetPutRequest};
 
 use crate::requests::cache::MomentoRequest;
 use crate::simple_cache_client::prep_request_with_timeout;
 use crate::{CacheClient, CollectionTtl, IntoBytes, MomentoResult};
 
+/// This trait defines an interface for converting a type into a vector of `SortedSetElement`s.
+pub trait IntoSortedSetElements: Send {
+    /// Converts the type into a vector of `SortedSetElement`s.
+    fn into_sorted_set_elements(self) -> Vec<SortedSetElement>;
+}
+
+/// Converts an iterator of value-score pairs into a vector of `SortedSetElement`s.
+///
+/// # Arguments
+///
+/// - `iter`: An iterator over pairs `(V, f64)` where `V` is a value that implements
+///   the `IntoBytes` trait, and `f64` represents the score associated with the value.
+///
+/// # Returns
+///
+/// A `Vec<SortedSetElement>` where each `SortedSetElement` contains a byte representation
+/// of the value and its associated score. The order of elements in the returned vector
+/// matches the order of pairs in the input iterator.
+///
+/// # Examples
+///
+/// Basic usage with a vector of tuples:
+///
+/// ```
+/// use momento::requests::cache::sorted_set::sorted_set_put_elements::map_and_collect_sorted_set_elements;
+/// let pairs = vec![("value1", 1.0), ("value2", 2.0)];
+/// let sorted_set_elements = map_and_collect_sorted_set_elements(pairs.into_iter());
+/// // `sorted_set_elements` is now a `Vec<SortedSetElement>` with byte representations of "value1" and "value2"
+/// # assert_eq!(sorted_set_elements, vec![
+/// #    momento_protos::cache_client::SortedSetElement {
+/// #        value: "value1".as_bytes().to_vec(),
+/// #        score: 1.0,
+/// #    },
+/// #    momento_protos::cache_client::SortedSetElement {
+/// #        value: "value2".as_bytes().to_vec(),
+/// #        score: 2.0,
+/// #    },
+/// # ]);
+/// ```
+///
+/// Usage with a `HashMap`:
+///
+/// ```
+/// use momento::requests::cache::sorted_set::sorted_set_put_elements::map_and_collect_sorted_set_elements;
+/// use std::collections::HashMap;
+/// let mut map = HashMap::new();
+/// map.insert("value1", 1.0);
+/// map.insert("value2", 2.0);
+/// let sorted_set_elements = map_and_collect_sorted_set_elements(map.into_iter());
+/// // `sorted_set_elements` is similar as above, suitable for sorted set operations
+/// # assert_eq!(sorted_set_elements, vec![
+/// #    momento_protos::cache_client::SortedSetElement {
+/// #        value: "value1".as_bytes().to_vec(),
+/// #        score: 1.0,
+/// #    },
+/// #    momento_protos::cache_client::SortedSetElement {
+/// #        value: "value2".as_bytes().to_vec(),
+/// #        score: 2.0,
+/// #    },
+/// # ]);
+/// ```
```

Comment:
these are great docs and I want to make sure we keep / expose them. I'm just hoping we can do it based on some flavor of `into` rather than the utility function. I am obviously missing something as to why that won't work, so I'll pull this branch down after I finish reviewing the PR and get myself up to speed :)